test/gutter_tree_test.cpp

```diff
@@ -14,87 +16,88 @@ static std::atomic<uint32_t> upd_processed;
   data_ret_t data;
   while(true) {
-    bool valid = buf_tree->get_data(data);
+    bool valid = gt->get_data(data);
     if (valid) {
       Node key = data.first;
       std::vector<Node> updates = data.second;
       // verify that the updates are all between the correct nodes
       for (Node upd : updates) {
         // printf("edge to %d\n", upd.first);
         ASSERT_EQ(nodes - (key + 1), upd) << "key " << key;
-        upd_processed += 1;
+        upd_processed++;
       }
     }
     else if(shutdown)
       return;
   }
 }
 
-void write_configuration(uint32_t buffer_exp, uint32_t fanout, int queue_factor, int page_factor) {
+void write_configuration(uint32_t buffer_exp, uint32_t fanout, int queue_factor, 
+                        int page_factor, int num_threads) {
   std::ofstream conf("./buffering.conf");
   conf << "buffer_exp=" << buffer_exp << std::endl;
   conf << "branch=" << fanout << std::endl;
   conf << "queue_factor=" << queue_factor << std::endl;
   conf << "page_factor=" << page_factor << std::endl;
+  conf << "num_threads=" << num_threads << std::endl;
 }
 
 // helper function to run a basic test of the buffer tree with
 // various parameters
 // this test only works if the depth of the tree does not exceed 1
 // and no work is claimed off of the work queue
 // to work correctly num_updates must be a multiple of nodes
-void run_test(const int nodes, const int num_updates, const uint32_t buffer_exp, const uint32_t branch_factor) {
-  printf("Running Test: nodes=%i num_updates=%i buffer_size 2^%i branch_factor %i\n",
+void run_test(const uint32_t nodes, const uint32_t num_updates, const int buffer_exp, const int branch_factor) {
+  printf("Running Test: nodes=%i num_updates=%i buffer_exp 2^%i branch_factor %i\n",
          nodes, num_updates, buffer_exp, branch_factor);
 
-  write_configuration(buffer_exp, branch_factor, 8, 1); // 8 is queue_factor, 1 is page_factor
-
-  BufferTree *buf_tree = new BufferTree("./test_", nodes, 1, true);
+  write_configuration(buffer_exp, branch_factor, 8, 1, 1); // 8 is queue_factor, 1 is page_factor, 1 is num_threads
+  GutterTree *gt = new GutterTree("./test_", nodes, 1, true); // 1 is number of workers
   shutdown = false;
   upd_processed = 0;
-  std::thread qworker(querier, buf_tree, nodes);
+  std::thread qworker(querier, gt, nodes);
 
-  for (int i = 0; i < num_updates; i++) {
+  for (unsigned i = 0; i < num_updates; i++) {
     update_t upd;
     upd.first = i % nodes;
     upd.second = (nodes - 1) - (i % nodes);
-    buf_tree->insert(upd);
+    gt->insert(upd);
   }
   printf("force flush\n");
-  buf_tree->force_flush();
+  gt->force_flush();
   shutdown = true;
-  buf_tree->set_non_block(true); // switch to non-blocking calls in an effort to exit
+  gt->set_non_block(true); // switch to non-blocking calls in an effort to exit
   qworker.join();
   ASSERT_EQ(num_updates, upd_processed);
-  delete buf_tree;
+  delete gt;
 }
 
 TEST(BasicInsert, Small) {
```

Comment:
Might wanna consider renaming each of these unit tests to GutterTreeTest for clarity.

Reply:
Good idea